libcontainer/cgroups/fs/memory.go

```diff
@@ -35,13 +40,10 @@ func (s *MemoryGroup) Apply(d *data) (err error) {
 		}
 	}()
 
-	if err := s.Set(path, d.c); err != nil {
-		return err
-	}
-
 	// We need to join memory cgroup after set memory limits, because
 	// kmem.limit_in_bytes can only be set when the cgroup is empty.
-	if _, err = d.join("memory"); err != nil {
+	_, err = d.join("memory")
```

Comment:
Earlier mkdir was called explicitly in line number 27, now via join again mkdir gets called. Looks like mkdir is called twice
is than redundant ?